### ISSUE TITLE
Clearing layers does not update the viewer

### ISSUE THREAD

**Issue by jni** [napari/napari#1829]
## 🐛 Bug

It looks like `layers.clear` is not connected to any viewer events, so the viewer is not aware when you clear layers programmatically, and it then becomes very unhappy when you try to do anything with a layer, because the layer doesn't exist anymore, except in the viewer's imagination. :joy: 

## To reproduce

```python
import numpy as np
import napari

with napari.gui_qt():
    v = napari.Viewer()
    v.add_image(np.random.random((6, 6)))
    v.add_image(np.random.random((6, 6)))
    v.layers.clear()
```

## Expected behavior

No layers should be visible.

## Environment

```
napari: 0.4.0rc2.dev3+g472f8109
Platform: Linux-4.15.0-29-generic-x86_64-with-glibc2.10
Python: 3.8.2 (default, Mar 26 2020, 15:53:00)  [GCC 7.3.0]
Qt: 5.14.2
PyQt5: 5.14.2
NumPy: 1.19.0rc1
SciPy: 1.4.1
Dask: 2.26.0
VisPy: 0.6.4

GL version:  3.0 Mesa 19.2.8
MAX_TEXTURE_SIZE: 16384

Plugins:
  - napari-plugin-engine: 0.1.6
  - ndtiffs: 0.1.1
  - svg: 0.1.4
```

**Comment by sofroniewn** [comment]
This is a duplicate of #1380, which would be fixed by #1504. I can look into #1504 again

**Comment by jni** [comment]
Ok, closing as duplicate.
